### PR TITLE
``v4`` Add ``ABIElementInfo`` to ``__init__.py``

### PR DESCRIPTION
**Pull request by reedsa**
### What was wrong?

``ABIElementInfo`` is missing from ``__init__.py``. It should be included as part of ``__all__``.

Docstrings in deprecated functions are missing deprecation warnings.
Deprecation messages were not entirely accurate and should list all relevant alternate types as part of the message.

### How was it fixed?

Add ``ABIElementInfo`` to ``__init__.py``.

Update deprecation messages to include all relevant new types, add deprecation messages to docstrings.

### Todo:

- [x] Clean up commit history

- [x] Add or update documentation related to these changes

- [x] Add entry to the [release notes](https://github.com/ethereum/eth-typing/blob/main/newsfragments/README.md)

#### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

**Review by pacrob — APPROVED** [review]
lgtm!

File: eth_typing/abi.py [hunk 1]
```diff
@@ -20,12 +20,26 @@
 Decodable = Union[bytes, bytearray]
 """Binary data to be decoded."""
 
-ABI_COMPONENT_DEPRECATION_MSG = "`{0}` is deprecated, use `ABIComponent` instead."
+ABI_COMPONENT_DEPRECATION_MSG = f"`{0}` is deprecated, use {1} instead."
```
Comment (kclowes) [inline eth_typing/abi.py:23]: It doesn't look like this is being used anywhere?

**Review by kclowes — COMMENTED** [review]
I am not seeing the warning when testing with ipython locally: 
```python
$ ipy
>>> from eth_abi import ABIFunctionParam
>>> ABIFunctionParam()
{}
```
 It may be a user error sort of deal, but worth a double check.

**Comment by reedsa** [comment]
I'm unable to replicate the issue, I am able to see the deprecation warning after I installed with `pip install git+https://github.com/reedsa/eth-typing@docstrings-with-deprecation-notice`.

File: eth_typing/abi.py (hunk 1 above)
Comment (reedsa) [reply eth_typing/abi.py]: oops, good catch!

**Review by kclowes — APPROVED** [review]
Okay, good. I don't know what's going on locally. Once docs pass, good to go!

**Comment by reedsa** [comment]
Circleci is giving me beef, connection drops from github for some reason. Going to keep kicking that job, will merge when it's green!

**Pull request merged**